src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -363,15 +364,20 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		this.uiController.initDialog(this._params && this._params.providers, this._model);
 	}
 
-	private handleFillInConnectionInputs(connectionInfo: IConnectionProfile): void {
-		this._connectionManagementService.addSavedPassword(connectionInfo).then(connectionWithPassword => {
+	private async handleFillInConnectionInputs(connectionInfo: IConnectionProfile): Promise<void> {
+		try {
+			const connectionWithPassword = await this.createModel(connectionInfo);
+
 			if (this._model) {
 				this._model.dispose();
 			}
-			this._model = this.createModel(connectionWithPassword);
-
+			this._model = connectionWithPassword;
 			this.uiController.fillInConnectionInputs(this._model);
-		}).catch(err => onUnexpectedError(err));
+		}
+		catch (err) {
+			onUnexpectedError(new Error(`Error filling in connection inputs with password. Original error message: ${err.message}`));
```

Comment:
We aren't guaranteed that this is an Error object, so using the message property is unsafe. Better to just get the string version directly (or use a helper function)

```suggestion
			onUnexpectedError(new Error(`Error filling in connection inputs with password. Original error message: ${err}`));
```